### ISSUE TITLE
Very slow offline deal acceptance on v1.5.1

### ISSUE THREAD

**Issue by hyunmoon** [filecoin-project/boost#1207]
### Checklist

- [X] This is **not** a question or a support request. If you have any boost related questions, please ask in the [discussion forum](https://github.com/filecoin-project/boost/discussions).
- [X] This is **not** a new feature or enhancement request. If it is, please open a [new idea discussion](https://github.com/filecoin-project/boost/discussions/new?category=ideas) instead. New feature and enhancement requests would be entertained by the boost team after a thorough discussion only.
- [X] I **have** searched on the [issue tracker](https://github.com/filecoin-project/boost/issues) and the [discussion forum](https://github.com/filecoin-project/boost/discussions), and there is no existing related issue or discussion.
- [X] I am running the [`Latest release`](https://github.com/filecoin-project/boost/releases), or the most recent RC(release canadiate) for the upcoming release or the dev branch(master), or have an issue updating to any of these.
- [X] I did not make any code changes to boost.

### Boost component

- [X] boost daemon - storage providers
- [ ] boost client
- [ ] boost UI
- [ ] boost data-transfer
- [ ] boost index-provider
- [ ] Other

### Boost Version

```text
boostd --version
boostd version 1.5.1+git.5aadde7

lotus-miner --version
lotus-miner version 1.19.0+mainnet+git.64059ca87

```

### Describe the Bug

**How to reproduce:**

On boost v1.5.1, using boost client, send an offline deal to your own boost daemon node.
The deal will take a very long time to be accepted (or rejected).
For me, each deal took about 15 seconds to be processed.

**How I fixed it:**

1. I downgraded boost and boostd to v1.5.0.
2. I also changed ConfigVersion = 4 to ConfigVersion = 3 in config.toml to downgrade.

After that, I could see a few dozens offline deals getting processed per second.

### Logging Information

```text
-
```

### Repo Steps

-

**Comment by hyunmoon** [comment]
I couldn't see any deals on v1.5.0 (lotus-miner not responding to an offline deal import)
So I upgraded to v1.5.1-rc5 and I get the following error:
`Error: creating node: starting node: starting storage provider: failed to list completed deals: scanning deal row: sql: Scan error on column index 6, name "FastRetrieval": sql/driver: couldn't convert <nil> (<nil>) into type bool`

**Comment by hyunmoon** [comment]
Everything works fine on boost v1.5.0 and lotus v1.8.0

**Comment by marco-storswift** [comment]
I meet the same issue. use the boost version is 1.6.0-rc1. the log info like this, it looks like processing deal acceptance request took too long times.
2023-02-20T21:46:36.236+0800    INFO    boost-net       lp2pimpl/net.go:197     received deal proposal  {"id": "be182211-ce0e-429b-b8f1-XXX", "client-peer": "12DXX"}
2023-02-20T21:46:36.236+0800    INFO    boost-storage-deal      logs/log.go:40  executing deal proposal received from network   {"id": "be182211-ce0e-429b-b8f1-XXX", "peer": "12DXX"}
2023-02-20T21:46:36.247+0800    INFO    boost-storage-deal      logs/log.go:40  processing deal acceptance request      {"id": "be182211-ce0e-429b-b8f1-XXX"}
2023-02-20T21:46:54.138+0800    INFO    boost-storage-deal      logs/log.go:40  offline deal accepted, waiting for data import  {"id": "be182211-ce0e-429b-b8f1-XXX"}

**Comment by LexLuthr** [comment]
At first glance, it could be because of https://github.com/filecoin-project/boost/pull/1067
Before v1.5.1-rc3, we were not applying deal filters to offline deals and they were directly added to the DB.

@hyunmoon @marco-storswift Can you confirm if you are using deal-filters?

**Comment by marco-storswift** [comment]
@hyunmoon  I used default config.toml
[LotusDealmaking]
  ConsiderOnlineStorageDeals = true
  ConsiderOfflineStorageDeals = true
  ConsiderOnlineRetrievalDeals = true
  ConsiderOfflineRetrievalDeals = true
  ConsiderVerifiedStorageDeals = true
  ConsiderUnverifiedStorageDeals = true
  PieceCidBlocklist = []
  ExpectedSealDuration = "24h0m0s"
  MaxDealStartDelay = "336h0m0s"
  PublishMsgPeriod = "1h0m0s"
  MaxDealsPerPublishMsg = 8
  MaxProviderCollateralMultiplier = 2
  MaxStagingDealsBytes = 0
  SimultaneousTransfersForStorage = 20
  SimultaneousTransfersForStoragePerClient = 0
  SimultaneousTransfersForRetrieval = 20
  StartEpochSealingBuffer = 480
  Filter = ""
  RetrievalFilter = ""
  [LotusDealmaking.RetrievalPricing]
    Strategy = "default"
    [LotusDealmaking.RetrievalPricing.Default]
      VerifiedDealsFreeTransfer = true
    [LotusDealmaking.RetrievalPricing.External]
      Path = ""

**Comment by beck-8** [comment]
> 

You don't see any problem here.

**Comment by dirkmc** [comment]
@LexLuthr I suspect the problem is the call to the lotus API to get the sectors summary:
https://github.com/filecoin-project/boost/blob/e8d1bedc53f67b2b43011a558ea61a8c657022ea/storagemarket/sealingpipeline/sealingpipeline.go#L25

Let's add some code to cache calls to `sealingpipeline.GetStatus`

**Comment by marco-storswift** [comment]
2023-02-20T21:46:### 36.247+0800 INFO boost-storage-deal logs/log.go:40 processing deal acceptance request {"id": "be182211-ce0e-429b-b8f1-XXX"}
2023-02-20T21:46:### 54.138+0800 INFO boost-storage-deal logs/log.go:40 offline deal accepted, waiting for data import {"id": "be182211-ce0e-429b-b8f1-XXX"}
one deal took 18 seconds

**Comment by hyunmoon** [comment]
I was not using any deal filters.

**Comment by beck-8** [comment]
> 2023-02-20T21:46:### 36.247+0800 INFO boost-storage-deal logs/log.go:40 processing deal acceptance request {"id": "be182211-ce0e-429b-b8f1-XXX"} 2023-02-20T21:46:### 54.138+0800 INFO boost-storage-deal logs/log.go:40 offline deal accepted, waiting for data import {"id": "be182211-ce0e-429b-b8f1-XXX"} one deal took 18 seconds

Sorry, I didn't notice the timing, 18 seconds is indeed problematic.I think anything more than a second is problematic.
![image](https://user-images.githubusercontent.com/34204218/220573515-5bc26f0f-e372-42fe-9d9d-03d67041b16b.png)

**Comment by marco-storswift** [comment]
> 

	summary, err := api.SectorsSummary(ctx)， to got sectors'list info will took too long time.

**Comment by beck-8** [comment]
This method is the same between v1.5.0 and v1.5.1. Didn't figure out why hyunmoon got better after the downgrade.

**Comment by hyunmoon** [comment]
To recall, I think v1.5.1-rc4 was free from the speed issue.

**Comment by marco-storswift** [comment]
@hyunmoon fix: run deal filter for offline deals (#1067) be added between  in v1.5.1-rc2 and v1.5.1-rc3.
@beck-8  I use the same boost version 1.6.0.rc1,  the miner's sector is very small, offline deal accepted very faster,  I test in big miner to run miner sector list, it tooked 16 seconds.  one deal offline deal accepted took 18 seconds.
But it look like it is very hard to fix this issue,

**Comment by hyunmoon** [comment]
I think you're right. The miner I was using had above 8PiB of Raw Byte Power.

**Comment by hyunmoon** [comment]
Use v1.5.1-rc1 for now then?

**Comment by dirkmc** [comment]
We are working on a fix, we should have something ready later today.

**Comment by beck-8** [comment]
In v1.6.0-rc1/rc2, I found that the speed of my boost to accept transactions became very slow, and it was accompanied by errors.
I don't think things are getting better, I don't know where it is making it worse.
my old version is v1.5.1
![img_v2_96001939-1c70-49ac-b493-e02083fc0aeh](https://user-images.githubusercontent.com/34204218/223082593-0a4780ae-1740-4030-afb4-d1a74c940296.jpg)

**Comment by beck-8** [comment]
https://github.com/filecoin-project/boost/issues/1222

**Comment by LexLuthr** [comment]
@beck-8 Did you try v1.6.0-rc3? Because this fix was not included in v1.6.0-rc2

**Comment by beck-8** [comment]
> @beck-8 Did you try v1.6.0-rc3? Because this fix was not included in v1.6.0-rc2

I'm trying to merge an update, can you point out which commit it is? Thanks

**Comment by LexLuthr** [comment]
https://github.com/filecoin-project/boost/releases/tag/v1.6.0-rc3

**Comment by beck-8** [comment]
which fix commit is it? which PR? I don't know which update here fixes this problem.

**Comment by LexLuthr** [comment]
Apologies for the confusion. I misunderstood the problem you described as a retrieval slowness.  The problem you are referring to was fixed in v1.6.0-rc2.
Since you have already updated to that version, can you tell me what the acceptance time looks like now for offline deals?

I am not sure what the attached screenshot is referring to here. You should check deal logs in UI to get logs related to a particular deal. Also, can you please get me the deal details and logs on this issue. Once we confirm the problem, I will be happy to reopen this.

**Comment by beck-8** [comment]
Let's move on here for discussion. I think it's more suitable. I'll add the details there later.
https://github.com/filecoin-project/boost/issues/1222
